File: client/src/lib/invoke/getBalance.ts
import { invoke } from "@tauri-apps/api/core";
import { ResultAsync } from "neverthrow";
import type { BalanceInfo, Config } from "@bindings";

export function getBalance(login: string, config: Config): ResultAsync<BalanceInfo, string> {
    return ResultAsync.fromPromise(
        invoke("get_balance", { login, config }),
        (error) => `Failed to get balance for login: ${error}`
    );
}

File: client/src-tauri/src/commands/get_balance.rs
use crate::config::Config;
use crate::nexium_api::get_balance as get_balance_api;
use crate::types::balance::BalanceInfo;

#[tauri::command]
pub async fn get_balance(
    login: String,
    config: Config,
) -> Result<BalanceInfo, String> {
    tauri::async_runtime::spawn_blocking(move || get_balance_api(login, config))
        .await
        .map_err(|e| format!("Failed to get balance: {}", e))?
}
